api/envoy/extensions/access_loggers/file/v3/file.proto

```diff
@@ -21,13 +21,31 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // Custom configuration for an :ref:`AccessLog <envoy_api_msg_config.accesslog.v3.AccessLog>`
 // that writes log entries directly to a file. Configures the built-in *envoy.access_loggers.file*
 // AccessLog.
-// [#next-free-field: 6]
+// [#next-free-field: 7]
 message FileAccessLog {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.config.accesslog.v2.FileAccessLog";
 
-  // A path to a local file to which to write the access log entries.
-  string path = 1 [(validate.rules).string = {min_len: 1}];
```

Comment:
The reason we don't wrap these two properties into a struct and make them `oneof` is to maintain backwards compatibility

Reply:
Stepping back for a sec, do you think it makes sense for a "file access log" to output to console, stderr, stdout, etc.? Or should we have ConsoleAccessLog, StderrAccessLog, StdoutAccessLog?

If we think putting this functionality in FileAccessLog makes sense, here is what I would suggest:

1) Deprecate path
2) Add a new oneof "output_type" or similar
3) Define a new message for each option: Path, Console, Stderr, Stdout (this allows for future extensibility)

I would then use ^ in the admin proto (and deprecate path their as well and/or potentially even allow loading full access log extensions so we can instantiate grpc access log, file access log, etc.). It's unfortunate that admin is not a full listener but that is a well known issue and I'm not suggesting we fix that here.

Reply:
I like the idea of creating `ConsoleAccessLog`, `StderrAccessLog`, `StdoutAccessLog` and using it everywhere.

Can I deprecate the path in `admin`  and replace it with a `config_type` that accepts on of the configs above? Or I can just add the config_type property in admin.

Reply:
What I would do is deprecate `path` in admin and actually make it a repeated list of access log extensions the same as we use elsewhere. You will see it will be trivial to wire up in the fake admin listener.